### PR TITLE
Copy resources to build dir

### LINKED ISSUE #31796: world_map.gpkg is changed everytime QGIS is run from build directory
This results in changes being reported in the local source directory and eventually clutters the commit history.

One way of solving this would be to focus on https://github.com/qgis/QGIS/issues/23991
But since this is probably a tough one to solve, I could imagine an alternative solution like copying the file to the build directory.

### PR DESCRIPTION
**Pull request by m-kuhn**
Fix #31796

## Description
Copies all resource files to the build directory during the build process. So the user can actively (or unintentionally) fiddle around with those files without git picking them up in the diff.

### CODE REVIEW

**Comment by nyalldawson** [comment]
Oh, nice move!!

File: resources/CMakeLists.txt [hunk 1]
```diff
@@ -1,36 +1,80 @@
...
+    "${CMAKE_BINARY_DIR}/output/resources/${RESOURCE_FILE}")
+ENDFOREACH(RESOURCE_FILE)
+
+# Install resources to system resource folder
+FOREACH(RESOURCE_FILE ${RESOURCES_FILES})
+  GET_FILENAME_COMPONENT(PATH_NAME "resources/${RESOURCE_FILE}" PATH)
+  INSTALL( FILES "resources/${RESOURCE_FILE}" DESTINATION ${PATH_NAME})
+ENDFOREACH()
+
+# srs.db -- spcial handling because there is a switch and rename depending on proj version
```
Comment (DelazJ) [inline resources/CMakeLists.txt]:
```suggestion
# srs.db -- special handling because there is a switch and rename depending on proj version
```

**Review by strk — APPROVED** [review]
I love it

**Comment by strk** [comment]
I didn't test if this also completely fixes #25830, deprecating #31980 -- will keep an eye and rebase when this is merged (Travis is happy so... go for it ?)

**Comment by strk** [comment]
This is the list of modified source files after `make check` with this PR as of commit e53af9fd4328fc835137a4fc4e32b229068f8d04:
```
        modified:   tests/testdata/curved_polys.gpkg
        modified:   tests/testdata/points_gpkg.gpkg
        modified:   tests/testdata/polys_overlapping_with_id.dbf
        modified:   tests/testdata/qgis_server/test_project_wms_grouped_layers.gpkg
        modified:   tests/testdata/qgis_server_accesscontrol/dimensions.gpkg
```

**Comment by m-kuhn** [comment]
It does not change the behavior regarding test data.
Let's merge it to master after branching 3.10 and possibly backporting down the road.

**Pull request merged**